### PR TITLE
Core Types

### PR DESCRIPTION
**Pull request by chippmann**

### CODE REVIEW

File: src/main/kotlin/godot/entrygenerator/extension/StringExt.kt [hunk 1]
```diff
@@ -1,5 +1,11 @@
 package godot.entrygenerator.extension
 
+fun String.camelToSnakeCase(): String {
```
Comment (piiertho) [inline src/main/kotlin/godot/entrygenerator/extension/StringExt.kt:3]: Note for the future, 
IIRC we have same funcgtion in entry generation, maybe have other methods that can be centralized.

**Review by piiertho — APPROVED** [review]

File: src/main/kotlin/godot/entrygenerator/extension/StringExt.kt (hunk 1 above)
Comment (chippmann) [reply src/main/kotlin/godot/entrygenerator/extension/StringExt.kt:3]: Jep we do have the same there but slightly different, so i opted for duplication. Does it really make sense to centralize such simple functions?

Comment (chippmann) [reply src/main/kotlin/godot/entrygenerator/extension/StringExt.kt:3]: If so I'll change it in this PR. I prefer sooner than later :-)

**Review by raniejade — APPROVED** [review]

**Comment by piiertho** [comment]
Oops, I merged core type without this PR merge. @chippmann can you merge this and update godot-jvm master ?

**Comment by chippmann** [comment]
> Oops, I merged core type without this PR merge. @chippmann can you merge this and update godot-jvm master ?

No worries. Sure I'll do that.

File: src/main/kotlin/godot/entrygenerator/extension/StringExt.kt (hunk 1 above)
Comment (piiertho) [reply src/main/kotlin/godot/entrygenerator/extension/StringExt.kt:3]: If it is not same impl maybe duplicate is better ^^

**Pull request merged**